lib/in_out/parser_regex.py

```diff
@@ -0,0 +1,80 @@
+log_regex = {
```

Comment:
This array seems quite like a configuration information. It may be better to read this into the program from a config file. The purpose of the config variables is not clear and it looks like there might be an overlap with the existing config variables in config.py. Please cross-check and if possible, move these to config.py.

Reply:
This is not exactly a config file. Config is something which has certain parameters which can be tweaked by a user but here we are trying to separately store a datastructure. I understand the intent in the comment but moving it to a config file, doesn't seem right to me, maybe we can rename this file. But if we still feel that this dictionary rightly belongs to config.py, please do let me know, I'll move it there then.

Reply:
I think it is fine to have it in a separate module and not config. Putting it in config will be misleading user suggesting that these are configurable and can be changed.

Reply:
@bhuvangupta008 In order to help me understand the issue better, please provide a comment on the fields of the groups JSON for different cases: nick_change, directed_messages etc.

Reply:
A regular expression defined as `r'={3} (\S+) is now known as (\S+)'` will match for a string like `=== foo is now know as bar`. when you run a `re.match` on the regex and the string a [matchObject](https://docs.python.org/2/library/re.html#match-objects) is returned.  
So from the given log line we need to extract 2 useful piece of information, the old nick and the new nick. In regex we can define groups. The groups would be marked by surrounding the parts of regular expression in parenthesis. The group numbering starts from 1 and goes onto the number of pairs of parenthesis. In the given example we have 2 groups defined (old_nick and the new_nick).  

The fields in groups.json is essentially a mapping between the group_no and the relevant tag for that group. For e.g. in our case it would be:  
- old_nick : 1
- new_nick : 2  
For further reading about groups, please check this [documentation](https://docs.python.org/2/howto/regex.html#grouping)

Reply:
The above description is useful and should be added as a comment to parser_regex.py module.